apps/projects/app/components/Shared/BountyContextMenu.js

```diff
@@ -17,110 +41,84 @@ const BountyContextMenu = ({ issue }) => {
-            <IconView color={`${theme.surfaceContent}`} />
-            <ActionLabel>
-              View work
-            </ActionLabel>
-          </Item>
-          <Item onClick={() => reviewApplication(issue)}>
-            <IconView color={`${theme.surfaceIcon}`} />
-            <ActionLabel>
-              View applications ({issue.requestsData.length})
-            </ActionLabel>
-          </Item>
-        </React.Fragment>
+      <MenuItem
+        panel={reviewApplication}
+        panelParams={issue}
+        caption={pluralize('View application', issue.requestsData.length)}
+        Icon={IconView}
+      />
+    </>
+  )
+  case 'in-progress': return (
+    <>
+      {(connectedAccount === assignee) && (
+        <MenuItem panel={submitWork} panelParams={issue} caption="Submit work" Icon={IconFile} />
       )}
-      {workStatus === 'funded' && (
-        <React.Fragment>
-          <Item onClick={() => requestAssignment(issue)}>
-            <IconFile color={`${theme.surfaceContent}`} />
-            <ActionLabel>
-              Submit application
-            </ActionLabel>
-          </Item>
-          {/* Disabled since the contract doesn't allow updating the amount */}
-          {/* <Item bordered onClick={() => editBounty([issue])}> */}
-          {/*   Update Funding */}
-          {/* </Item> */}
-        </React.Fragment>
+      <MenuItem
+        panel={reviewApplication}
+        panelParams={issue}
+        caption={pluralize('View application', issue.requestsData.length)}
+        Icon={IconView}
+      />
+    </>
+  )
+  case 'review-work': return (
+    <>
+      {openSubmission && (
+        <MenuItem panel={submitWork} panelParams={issue} caption="Submit work" Icon={IconFile} />
       )}
-      {workStatus === 'review-applicants' && (
-        <React.Fragment>
-          {!pastDeadline && (
-            <Item onClick={() => requestAssignment(issue)}>
-              <IconFile color={`${theme.surfaceContent}`} />
-              <ActionLabel>
-                Submit application
-              </ActionLabel>
-            </Item>
-          )}
-          <Item onClick={() => reviewApplication(issue)}>
-            <IconView color={`${theme.surfaceContent}`} />
-            <ActionLabel>
-              View applications ({issue.requestsData.length})
-            </ActionLabel>
-          </Item>
-          {/* Disabled since the contract doesn't allow updating the amount */}
-          {/*<Item bordered onClick={() => editBounty([issue])}>*/}
-          {/*  <IconCoin color={`${theme.surfaceContent}`} />*/}
-          {/*  <ActionLabel>*/}
-          {/*    Update Funding*/}
-          {/*  </ActionLabel>*/}
-          {/*</Item>*/}
-        </React.Fragment>
+      <MenuItem
+        panel={reviewWork}
+        panelParams={issue}
+        caption={pluralize('View work submission', issue.workSubmissions.length)}
+        Icon={IconView}
+      />
+      {openSubmission || (
```

Comment:
Does this `||` work as expected? Do we end up with `true` in the UI sometimes?

I think I'd prefer we rephrased this as `!openSubmission &&`

Reply:
It works as expected, but sure, I changed it to more standard `!openSubmission &&`.